### PR TITLE
OSASINFRA-3485: OpenStack Node pools

### PR DESCRIPTION
**Pull request by EmilienM**

### CODE REVIEW

**Comment by EmilienM** [comment]
@MaysaMacedo @mdbooth please review

File: hypershift-operator/controllers/hostedcluster/internal/platform/openstack/openstack.go [hunk 1]
```diff
@@ -331,3 +337,119 @@ func (a OpenStack) DeleteCredentials(ctx context.Context, c client.Client, hclus
...
+
+	// Rules for worker security group with the remote IP prefix set to the machine CIDRs
+	for _, machineCIDR := range machineCIDRs {
+		machineCIDRInboundRules := []capo.SecurityGroupRuleSpec{
+			{
+				Name:     "esp-ingress",
+				Protocol: k8sutilspointer.String("esp"),
+			},
+			{
+				Name:     "icmp-ingress",
```
Comment (MaysaMacedo) [inline hypershift-operator/controllers/hostedcluster/internal/platform/openstack/openstack.go:367]: shouldn't ping be allowed from everywhere?

File: hypershift-operator/controllers/hostedcluster/internal/platform/openstack/openstack.go [hunk 2]
```diff
@@ -331,3 +337,119 @@ func (a OpenStack) DeleteCredentials(ctx context.Context, c client.Client, hclus
...
+				Protocol: k8sutilspointer.String("vrrp"),
+			},
+		}
+
+		for i, rule := range machineCIDRInboundRules {
+			rule.RemoteIPPrefix = k8sutilspointer.String(machineCIDR)
+			machineCIDRInboundRules[i] = rule
+		}
+
+		ingressRules = append(ingressRules, machineCIDRInboundRules...)
```
Comment (MaysaMacedo) [inline hypershift-operator/controllers/hostedcluster/internal/platform/openstack/openstack.go:393]: the rule for port 10250, which is for kubelet insecure and 30000-32767 for ingress services are already created by default by CAPO  right?

**Review by MaysaMacedo — APPROVED** [review]
In general LGTM, just have two questions

File: hypershift-operator/controllers/hostedcluster/internal/platform/openstack/openstack.go (hunk 1 above)
Comment (EmilienM) [reply hypershift-operator/controllers/hostedcluster/internal/platform/openstack/openstack.go:367]: AWS rules in HCP:

API 	 6443 	     tcp machine CIDRs for private link endpoint
API 	 6443 	     udp machine CIDRs for private link endpoint
Geneve 	 6081 udp
HTTPS 	 443 	     tcp machine CIDRs
HTTPS 	 443 	     udp machine CIDRs
ICMP                 machine CIDRs
IKE 	 500 udp
IKE NAT	 4500 udp
Internal 9000 9999   tcp
Internal 9000 9999   udp
Kubelet  10250       tcp
Nodeport 30000 32767 tcp
Nodeport 30000 32767 udp
SSH 	 22 	     tcp machine CIDRs
VXLAN 	 4789 udp

Honestly I don't think allowing ICMP from anywhere is a good idea but we can relax it later eventually.

File: hypershift-operator/controllers/hostedcluster/internal/platform/openstack/openstack.go (hunk 2 above)
Comment (EmilienM) [reply hypershift-operator/controllers/hostedcluster/internal/platform/openstack/openstack.go:393]: * 10250: https://github.com/kubernetes-sigs/cluster-api-provider-openstack/blob/20e8bac7adbbced3949d92c8d34850b35a28dd4a/pkg/cloud/services/networking/securitygroups_rules.go#L76-L98
* 30000-32767: https://github.com/kubernetes-sigs/cluster-api-provider-openstack/blob/20e8bac7adbbced3949d92c8d34850b35a28dd4a/pkg/cloud/services/networking/securitygroups_rules.go#L145-L164

**Pull request merged**